xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -757,6 +756,8 @@ private void updateRoutes(List<VirtualHost> virtualHosts, long httpMaxStreamDura
           }
           ClientInterceptor filters = createFilters(filterConfigs, virtualHost, route, null);
           routesData.add(new RouteData(route.routeMatch(), route.routeAction(), filters));
+        } else {
+          // Discard route
```

Comment:
You explained me why we do this IRL, but your intention here may not be clear for other people reading this.
I recommend adding a note here that it mirrors the logic in `selectConfig`, while making this clear that here in `updateRoutes` we intentionally discarding (as opposed to throwing in `selectConfig`).

Reply:
The comment here means we are intentionally discarding... I added a comment to selectConfig that this method discards unknown actions.